Fixing K-type RGB lighting

Andrew-Fahmy opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->
I was able to fix the RGB lighting on the K-type which before would only light up half the keyboard. 

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description
The K-type has two IS31FL3733 chips which are on different busses, and the issue was that the i2c_master driver was hard coded to use only one bus. The way I fixed it way by copying both the IS31FL3733 and ic2_master drivers into the local folder and making the changes there so the original core drivers are left unchanged.   

I would like to note that the RGB lighting has a significant effect on the performance of the keyboard so in order to keep the matrix scan frequency above 1000 I have reduced the lighting animations to 10fps. The matrix scan frequency may still drop below 1000 on intensive animations such as the key reactive animations. The raindrop animation does not suffer from the issue as it only updates one LED at a time, 

<!--- Describe your changes in detail here. -->

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

tzarc commented:
Haven't had a chance to do a comprehensive review, but I just want to thank you for doing a bunch of the legwork on the multi-I2C-bus driver. It's been on my TODO list for quite some time!

drashna commented on keyboards/k_type/keymaps/andrew-fahmy/config.h:
```diff
@@ -0,0 +1,34 @@
+#pragma once
```
Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

drashna commented on keyboards/k_type/keymaps/andrew-fahmy/keymap.c:
```diff
@@ -0,0 +1,44 @@
+#include QMK_KEYBOARD_H
```
Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

Andrew-Fahmy commented:
I have went through and added GLP licenses to the files that need it. I also refactored my code so that it is maintainable, and can be build off in the future to work with any number of i2c busses.

fauxpark commented:
I would have preferred this be integrated into core...
